src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,393 @@
+ * process. This is particularly important on slow network connections.
+ *
+ * @deprecated Will be made private in v5.0
+ */
+ExitThisPage.prototype.exitPage = function (redirectUrl) {
+  this.$updateSpan.innerText = ''
+
+  // Blank the page
+  // As well as creating an overlay with text, we also set the body to hidden
+  // to prevent screen reader and sequential navigation users potentially
+  // navigating through the page behind the overlay during loading
+  document.body.classList.add('govuk-exit-this-page-hide-content')
+  this.$overlay = document.createElement('div')
+  this.$overlay.className = 'govuk-exit-this-page-overlay'
+  this.$overlay.setAttribute('role', 'alert')
+
+  // we do these this way round, thus incurring a second paint, because changing
+  // the element text after adding it means that screen readers pick up the
+  // announcement more reliably.
+  document.body.appendChild(this.$overlay)
+  this.$overlay.innerText = this.i18n.t('activated')
+
+  window.location.href = this.$button.getAttribute('href')
+}
+
+/**
+ * Pre-activation logic for when the button is clicked/activated via mouse or
+ * pointer.
+ *
+ * We do this to differentiate it from the keyboard activation event because we
+ * need to run `e.preventDefault` as the button or skiplink are both links and we
+ * want to apply some additional logic in `exitPage` before navigating.
+ *
+ * @deprecated Will be made private in v5.0
+ * @param {MouseEvent} e - mouse click event
+ */
+ExitThisPage.prototype.handleClick = function (e) {
+  e.preventDefault()
+  this.exitPage()
+}
+
+/**
+ * Logic for the 'quick escape' keyboard sequence functionality (pressing the
+ * Shift key three times without interruption, within a time limit).
+ *
+ * @deprecated Will be made private in v5.0
+ * @param {KeyboardEvent} e - keyup event
+ */
+ExitThisPage.prototype.handleKeypress = function (e) {
+  // Detect if the 'Shift' key has been pressed. We want to only do things if it
+  // was pressed by itself and not in a combination with another key—so we keep
+  // track of whether the preceding keyup had shiftKey: true on it, and if it
+  // did, we ignore the next Shift keyup event.
+  //
+  // This works because using Shift as a modifier key (e.g. pressing Shift + A)
+  // will fire TWO keyup events, one for A (with e.shiftKey: true) and the other
+  // for Shift (with e.shiftKey: false).
+  if (
+    (e.key === 'Shift' || e.keyCode === 16 || e.which === 16) &&
+    !this.lastKeyWasModified
+  ) {
+    this.keypressCounter += 1
+
+    // Update the indicator before the below if statement can reset it back to 0
+    this.updateIndicator()
+
+    // Clear the timeout for the keypress timeout message clearing itself
+    if (this.timeoutMessageId !== null) {
+      clearTimeout(this.timeoutMessageId)
+      this.timeoutMessageId = null
+    }
+
+    if (this.keypressCounter >= 3) {
+      this.keypressCounter = 0
+
+      clearTimeout(this.keypressTimeoutId)
```

Comment:
For the timeout `this.timeoutMessageId` we did a "not null" check first

Do we need one here?

Reply:
Although we clear timers and intervals without checking in a few places, so maybe we're being [too strict above](https://github.com/alphagov/govuk-frontend/pull/2545/files#diff-73815616f3d4872193e2c150dfa54c35ee4232882d6b0441f749c833636a3d56R234)